File: tasks/typescript/8kyu/JavaScriptArrayFilter.ts
// JavaScript Array Filter

// The solution would work like the following:

// getEvenNumbers([2,4,5,6]) // should == [2,4,6]

// Sample Tests:
// import { assert, config } from "chai";

// import { getEvenNumbers } from "./solution";

// config.truncateThreshold = 0

// describe("Fixed tests", function() {
//   it("should return an empty array if there are no even numbers", function() {
//       assert.deepEqual(getEvenNumbers([1,2,3,6,8,10]), [2,6,8,10])
//       assert.deepEqual(getEvenNumbers([1,2]), [2])
//       assert.deepEqual(getEvenNumbers([12,14,15]), [12,14])
//       assert.deepEqual(getEvenNumbers([13,15]), [])
//       assert.deepEqual(getEvenNumbers([1,3,9]), [])
//   });
// });

export const getEvenNumbers = (numbersArray: number[]): number[] => {
  return numbersArray.filter(x => !(x % 2));
};

File: tasks/cpp/7kyu/JavaScriptArrayFilter.cpp
/*
The solution would work like the following:

get_even_numbers({2,4,5,6}) => {2,4,6}
#include <vector>

Describe(Fixed_tests) {
  It(Base_cases) {
    Assert::That(get_even_numbers(std::vector<int>{2, 4, 5, 6}), 
                 Equals(std::vector<int>{2, 4, 6}));
    
    Assert::That(get_even_numbers(std::vector<int>{}), 
                 Equals(std::vector<int>{}));
    
    Assert::That(get_even_numbers(std::vector<int>{1}), 
                 Equals(std::vector<int>{}));
    
    Assert::That(get_even_numbers(std::vector<int>{1, 2}), 
                 Equals(std::vector<int>{2}));
    
    Assert::That(get_even_numbers(std::vector<int>{1, 2, 3, 4, 5}), 
                 Equals(std::vector<int>{2, 4}));
    
    Assert::That(get_even_numbers(std::vector<int>{2, 4, 6, 8}), 
                 Equals(std::vector<int>{2, 4, 6, 8}));
  }
};
*/

#include <vector>

std::vector<int> get_even_numbers(const std::vector<int>& arr) {
  std::vector<int> answ;
  for (int i : arr) {
    if (i % 2 == 0) {
      answ.push_back(i);
    }
  }
  return answ;
}
